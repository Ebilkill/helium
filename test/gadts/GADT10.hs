data Expr a where
    I :: Int -> Expr Int
    B :: Bool -> Expr Bool
    Plus :: Num a => Expr a -> Expr a -> Expr a
    Equals :: Eq a => Expr a -> Expr a -> Expr Bool

eval :: Expr a -> a
eval (I i) = i
eval (B b) = b
eval (Plus x y) = eval x + eval y
eval (Equals x y) = eval x == eval y

-- 3 + 4 == 7
answer1 = eval (Equals (Plus (I 3) (I 4)) (I 7))