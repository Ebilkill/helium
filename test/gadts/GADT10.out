Prelude is up to date
Compiling gadts/GADT10.hs
fromList [(answer1,Bool),(eval,Expr a -> a)]

Length of errors doesn't match

fromList [(answer1,Bool),(eval,Expr a -> a)]
"fromList [(answer1,Bool),(eval,Expr a -> a)]"
(8,7): Type error in pattern application
 pattern          : I i
 constructor      : I
   type           : Int  -> Expr Int 
   does not match : Bool -> Expr Bool

Compilation failed with 1 error
